File: back/src/main/java/com/quest/buddy/models/Sport.java
package com.quest.buddy.models;

import javax.persistence.*;

import com.quest.buddy.dtos.SportDto;

import org.hibernate.annotations.Formula;
import org.modelmapper.ModelMapper;

@Entity
@Table(name = "sport")
public class Sport implements BaseModel<SportDto> {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "name", length = 100, nullable = false)
    private String name;


    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "icon", length = 500)
    private String icon;

    @Column(name ="active", nullable = false)
    private Boolean active = true;

    @Formula("(select count(e.id) from Event e where e.sport_id = id)")
    private Long nbEvents;

    @Formula("(select count(s.id) from my_sport s where s.sport_id = id)")
    private Long nbUsers;

    public Sport() {
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public Boolean getActive() {
        return active;
    }

    public void setActive(Boolean active) {
        this.active = active;
    }

    public Long getNbEvents() {
        return nbEvents;
    }

    public void setNbEvents(Long nbEvents) {
        this.nbEvents = nbEvents;
    }
    public Long getNbUsers() {
        return nbUsers;
    }
    public void setNbUsers(Long nbUsers) {
        this.nbUsers = nbUsers;
    }
    @Override
    public SportDto toDto(){
        ModelMapper modelMapper = new ModelMapper();
        SportDto sportDto = modelMapper.map(this, SportDto.class);
        return sportDto;
    }

}

File: front/src/app/models/sport.ts
export class Sport{
  constructor(private _id:number,
  private _name:string,
  private _description:string,
  private _icon:string,
  private _active:boolean,
  private _nbEvents:number,
  private _nbUsers:number) {
  }

  get id(): number {
    return this._id;
  }

  set id(value: number) {
    this._id = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get description(): string {
    return this._description;
  }

  set description(value: string) {
    this._description = value;
  }

  get icon(): string {
    return this._icon;
  }

  set icon(value: string) {
    this._icon = value;
  }

  get active(): boolean {
    return this._active;
  }

  set active(value: boolean) {
    this._active = value;
  }

  get nbEvents(): number {
    return this._nbEvents;
  }

  set nbEvents(value: number) {
    this._nbEvents = value;
  }

  get nbUsers(): number {
    return this._nbUsers;
  }

  set nbUsers(value: number) {
    this._nbUsers = value;
  }
}
